packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -268,10 +336,23 @@ class OnsetPicker extends React.Component {
     if (this.props.onsetOnChange === undefined) {
       return;
     }
-
+    const birthDate = moment(this.props.birthdate);
+    // Set onset date to 1st day of next month of birthdate if onset date is less than birthdate.
+    if (this.state.onsetDate && moment(this.state.onsetDate) < birthDate) {
+      this.setState(() => ({
+        onsetDate: birthDate.month() !== 11 ? moment().year(birthDate.year()).month(birthDate.month() + 1).startOf('month') : moment().year(birthDate.year() + 1).startOf('year'),
```

Comment:
we might invoke `birthDate.month()`twice. Can we invoke once and save that value into a local variable for re-use before the setState call? We could also apply the same change to `birthDate.year()` for consistency and make this line a tad easier to read :D
```js
const yearOfBirthData = birthDate.year();
const monthOfBirthDate = birthDate.month();
this.setState(() => ({
  onsetDate: monthOfBirthDate !== 11 ? moment().year(yearOfBirthDate).month(monthOfBirthDate + 1).startOf('month') : moment().year(yearOfBirthDate + 1).startOf('year'),
```

Reply:
Added variables ee819ec